apollo-backend/src/test/java/io/logz/apollo/helpers/ModelsGenerator.java

```diff
@@ -180,7 +182,15 @@ public static Deployment createAndSubmitDeployment(ApolloTestClient apolloTestCl
 
         // Now we have enough to create a deployment
         Deployment testDeployment = ModelsGenerator.createDeployment(testService, testEnvironment, testDeployableVersion);
-        testDeployment.setId(apolloTestClient.addDeployment(testDeployment).getId());
+        MultiDeploymentResponseObject result = apolloTestClient.addDeployment(testDeployment);
+
+        try {
+            LinkedHashMap deployment = (LinkedHashMap) result.getSuccessful().get(0).get("deployment");
+            testDeployment.setId((int) deployment.get("id"));
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
This is not a valid exception to catch... you can use an "if" to see if there are any successful ones. not try to get it and catch this exception

Reply:
Also, now that this have logic, just call the overload below here